gh-docs/synapse-unit-testing-framework/synapse-unit-testing-framework.md

```diff
@@ -0,0 +1,375 @@
+# Synapse Unit Testing Framework
+Unit testing framework for synapse provides an in-built way to execute a unit test against a given synapse configuration artifacts such as sequence, proxy, api, local entry, endpoint. Mainly it required a test artifact which is your main configuration, supportive artifacts which needs to execute the test artifact if exists and test cases with assertions. 
+
+### Features
+- Support deploy sequence, proxy, API, endpoint and local-entry artifacts
+- Able to run multiple test cases against a test artifact and supportive artifacts
+- Able to create, start and stop mock-services with multiple resources
+- Assertion with `AssertEquals` and `AssertNotNull` properties
+- Assertion for response body(`$body`), message context property(`$ctx`), axis2 property(`$axis2`) and transport property(`$trp`) values
+
+### Running Unit Testing Framework
+
+You can use the following command depending on the platform to start the unit testing framework in the Enterprise Integrator.
+- MacOS/Linux/CentOS - `sh <MI_HOME>/bin/integrator.sh -synapseTest <port>`
+- Windows - `<MI_HOME>/bin/integrator.bat -synapseTest <port>`
+
+By default the unit testing framework starts on port 9007 and you can change the starting port by passing the port as above
+
+### Stopping Unit Testing Framework
+To stop the Unit Testing Framework runtime, press Ctrl+C in the command window and it will stop the server along with the Enterprise Integrator runtime.
+
+### How To Send a TCP Request
+
+You can use following code segment to send a sample test case to the unit testing framework and get the server response.
+```
```

Comment:
Shall we use language-specific syntax highlighting here? i.e. 
```
```java
```

Reply:
Added